src/Engine/Data/Texture.cpp

```diff
@@ -94,7 +95,83 @@ void Texture::bindImageTexture( int unit,
         uint( unit ), level, layered, layer, access, m_textureParameters.internalFormat );
 }
 
-void Texture::updateData( const void* data ) {
+void Texture::updateData( void* newData ) {
```

Comment:
Why removing the const qualifier ?

Reply:
Because textureParameters.textels is void*, and could be update (no const).
There is a comment (Texture.hpp:79) says that the pointer is dangling and that it is up to the user to maintain the data. Risk of simultaneous reading and writing if the floating pointer is updated by the user at the same time as the updateGL. Should we improve this by adding an updateData function that copies the user data internally for future synchronization without simultaneous read/write in the image buffer ?

Reply:
For information textureParameters.texels is now the internal representation of pixel data of Texture. Before this contribution updateData perform the gpu synchronization on the fly with random pointer and not the internal pointer.

Reply:
The updateData only update internal data. No updateGL is performed here within this function.